tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +310,17 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					const avgHeapUsed = heapUsedStats.arithmeticMean;
+					const allowedMemoryUsage =
+						args.baselineMemoryUsage * (1 + args.allowedDeviation / 100);
+					const tolerance = args.baselineMemoryUsage * (args.allowedDeviation / 100);
+
+					if (ENABLE_MEM_REGRESSION && avgHeapUsed > allowedMemoryUsage) {
+						throw new Error(
+							`Memory Regression detected for ${testObject.title}: Used ${avgHeapUsed} bytes, exceeding the baseline of ${args.baselineMemoryUsage} bytes., with an allowed tolerance of ${tolerance} bytes.`,
```

Comment:
Nit: I think leaving the tolerance as a percent for the error is fine. The exact number of bytes won't matter much.